examples/AdvancedOperations/AddAppCampaign.php

```diff
@@ -0,0 +1,443 @@
+            // Creates a campaign criterion.
+            $campaignCriterion = new CampaignCriterion([
+                'campaign' => new StringValue(['value' => $campaignResourceName]),
+                'type' => CriterionType::LANGUAGE,
+                'language' => new LanguageInfo([
+                    'language_constant' => new StringValue(['value' =>
+                        ResourceNames::forLanguageConstant($languageId)
+                    ])
+                ])
+            ]);
+
+            // Creates a campaign criterion operation.
+            $campaignCriterionOperation = new CampaignCriterionOperation();
+            $campaignCriterionOperation->setCreate($campaignCriterion);
+
+            $campaignCriterionOperations[] = $campaignCriterionOperation;
+        }
+
+        // Submits the criteria operations.
+        $campaignCriterionServiceClient = $googleAdsClient->getCampaignCriterionServiceClient();
+        $response = $campaignCriterionServiceClient->mutateCampaignCriteria(
+            $customerId,
+            $campaignCriterionOperations
+        );
+
+        $createdCampaignCriteriaResourceName = $response->getResults()[0]->getResourceName();
+        printf("Created Campaign Criteria '%s'%s", $createdCampaignCriteriaResourceName, PHP_EOL);
+    }
+
+    /**
+     * Creates an ad group and associated app ap for a given campaign.
+     *
+     * @param GoogleAdsClient $googleAdsClient the Google Ads API client
+     * @param int $customerId the customer ID
+     * @param string $campaignResourceName the resource name of the campaign to be modified
+     */
+    private static function createAppAd(
+        GoogleAdsClient $googleAdsClient,
+        int $customerId,
+        string $campaignResourceName
+    ) {
+        // Creates an ad group.
+        // Note that the ad group type must not be set.
+        // Since the advertising_channel_sub_type is APP_CAMPAIGN,
+        //   1- you cannot override bid settings at the ad group level.
+        //   2- you cannot add ad group criteria
+        $adGroup = new AdGroup([
+            'name' => new StringValue(['value' => 'Earth to Mars cruises ' . uniqid()]),
+            'status' => AdGroupStatus::ENABLED,
+            'campaign' => new StringValue(['value' => $campaignResourceName])
+        ]);
+
+        // Creates an ad group operation.
+        $adGroupOperation = new AdGroupOperation();
+        $adGroupOperation->setCreate($adGroup);
+
+        // Submits the ad group operation to add the ad group.
+        $adGroupServiceClient = $googleAdsClient->getAdGroupServiceClient();
+        $response = $adGroupServiceClient->mutateAdGroups($customerId, [$adGroupOperation]);
+
+        $createdAdGroupResourceName = $response->getResults()[0]->getResourceName();
+        printf(
+            "Ad Group created '%s'%s",
+            $createdAdGroupResourceName,
+            PHP_EOL
+        );
+
+        // Creates an ad group ad.
+        $adGroupAd = new AdGroupAd([
```

Comment:
Could you split the method into two -- one for adgroup creation and the other for ad creation? The method here is named createAppAd, but it creates an adgroup and an ad, which doesn't sound right.

Reply:
I agree. @fiboknacky this sounds like a major change to me, I am happy to do it with PHP but not sure that it is the best process to follow: I think the golden needs to be updated first?

Reply:
I agree. @fiboknacky this sounds like a major change to me, I am happy to do it with PHP but not sure that it is the best process to follow: I think the golden needs to be updated first?

Reply:
For this one, the reason why I didn't speak up (actually I noticed it too) was I thought that probably the golden CL author would want this to match the guide, which is also divided into two parts: 1) creating a campaign, 2) creating an ad group and ad.

But I see no harm to do what Anash suggested too.
For this one, we would need to reach out to the golden CL author (and other libs that already implemented it) to change too.

Reply:
@wihl WDYT?

Reply:
Yes @fiboknacky is correct. This was done explicitly to mirror the guide (as were many of the comments). As the guides reference the examples, the examples should IMO be consistent with the guides.

Reply:
Yes @fiboknacky is correct. This was done explicitly to mirror the guide (as were many of the comments). As the guides reference the examples, the examples should IMO be consistent with the guides.

Reply:
Yes @fiboknacky is correct. This was done explicitly to mirror the guide (as were many of the comments). As the guides reference the examples, the examples should IMO be consistent with the guides.

Reply:
SGTM. @AnashOommen you OK with no split?

Reply:
@AnashOommen I did the change as discussed offline.
@fiboknacky I will notify the other client lib owners of the change as soon as it is reviewed and merged.